bundles/org.eclipse.passage.lic.base/src/org/eclipse/passage/lic/base/agreements/AgreementAssessmentService.java

```diff
@@ -36,15 +37,14 @@ public AgreementAssessmentService(Collection<Requirement> requirements, Agreemen
 
 	public Collection<AgreementToAccept> assessment() {
 		return requirements.stream()//
-				.map(this::assessment)//
-				.flatMap(Collection::stream)//
+				.flatMap(this::assessment)//
 				.collect(Collectors.toList());
 	}
 
-	private Collection<AgreementToAccept> assessment(Requirement requirement) {
+	private Stream<AgreementToAccept> assessment(Requirement requirement) {
```

Comment:
Returning a `Stream` is a dangerous practice. The reason is that it is not a real daya type. It's a sort of a _data flow_ and it is _severely stateful_. Generally speaking, you never know in which state you've got a stream right now and whether or not it will be the same next time. 

Thus it's practically forbidden for public and package local methods and must be avoided in private too, unless you have good reasons to use it (like using _try-with-resource_ for a stream, that is complicated to create)

There is no performance issue here. Code quality must not be deminished unless it is proven that 
 - there is indeed a performapce problem 
 -  there is no other way to  address it

Reply:
I agree with your arguments and your statement regarding returning Streams in public/package-private methods, but for private methods in a small scope I would have accepted it, but I also understand if you want to avoid it.
-> Reverted that part.